cli/cmd/common/diff.go

```diff
@@ -0,0 +1,144 @@
+package common
+
+import (
+	"reflect"
+	"sort"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+)
+
+// Diff returns a string representation of the difference between two maps as YAML.
+// The returned string is sorted alphabetically by key. If the maps are identical, the returned string is empty.
+func Diff(a, b map[string]interface{}) (string, error) {
+	if len(a) == 0 && len(b) == 0 {
+		return "", nil
+	}
+
+	buf := new(strings.Builder)
+	err := diffRecursively(a, b, 0, buf)
+
+	return buf.String(), err
+}
+
+// diffRecursively iterates over both maps and writes the differences to the given buffer.
+func diffRecursively(a, b map[string]interface{}, recurseDepth int, buf *strings.Builder) error {
+	// Get the union of keys in a and b sorted alphabetically.
```

Comment:
Just for consistency of output. Otherwise the diff could come out in a different order every time. I figure alphabetically is a fair enough consistent sorting. 

With some additional work, I could get it outputting in the same order as values.yaml, but it didn’t seem like the right choice for now.

Without some sorting it would be harder (not impossible) to test as well.

Reply:
Oh got it, thanks for clarifying that.